sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsScaleMonitor.cs

```diff
@@ -130,24 +107,22 @@ internal async Task<EventHubsTriggerMetrics> CreateTriggerMetrics(List<Partition
             List<string> partitionErrors = new List<string>();
             for (int i = 0; i < partitionRuntimeInfo.Count; i++)
             {
-                Tuple<BlobParitionCheckpoint, string> partitionLeaseFile = await GetPartitionLeaseFileAsync(i).ConfigureAwait(false);
-                BlobParitionCheckpoint partitionLeaseInfo = partitionLeaseFile.Item1;
-                string errorMsg = partitionLeaseFile.Item2;
+                var partitionProperties = partitionRuntimeInfo[i];
 
-                if (partitionRuntimeInfo[i] == null || partitionLeaseInfo == null)
+                var checkpoint = (BlobsCheckpointStore.BlobEventProcessorCheckpoint)checkpoints.SingleOrDefault(c => c.PartitionId == partitionProperties.Id);
```

Comment:
Unless you're using a very aggressive checkpointing strategy, which I would not recommend, this isn't going to be your most accurate indicator of what events have been processed.   Have you considered holding onto the last event that had been processed and using that as an indicator?  

Bonus:  If you turn on the option for `TrackLastEnqueuedEventProperties`  for the processor, it will populate each event with the same information that you're querying with the partition properties rather than making an active call (which will be throttled if you're doing it too frequently).    You'll lose a small bit of fidelity here since you'd see a snapshot rather than real-time, but probably would still have more accuracy than the current checkpoint-based approach.